catalog/task/create-internal-request/0.4/create-internal-request.yaml

```diff
@@ -0,0 +1,120 @@
+spec:
+  description: >-
+      Creates an InternalRequest resource to call IIB service
+  params:
+    - name: pipelineRunName
+      type: string
+      description: The name of the Parent PipelineRun of this task
+    - name: binaryImage
+      type: string
+      description: FBC Image to be added to index image
+    - name: fbcFragment
+      type: string
+      description: FBC fragment built by HACBS
+    - name: fromIndex
+      type: string
+      description: Index image (catalog of catalogs) the FBC fragment will be added to    
+    - name: overwriteFromIndex
+      type: string
+      description: Boolean indicating if the fromIndex should be overwritten
+    - name: buildTags
+      type: string
+      description: List of additional tags the internal index image copy should be tagged with
+    - name: addArches
+      type: string
+      description: List of arches the index image should be built for
+    - name: requestUpdateTimeout
+      type: string
+      default: "360"
+      description: Max seconds waiting for the status update
+    - name:  buildTimeoutSeconds
+      type: string
+      default: "300"
+      description: Max seconds waiting for the status update
+  results:
+    - name: requestMessage
+      description: Internal Request message
+    - name: requestReason
+      description: Internal Request reason
+    - name: requestResults
+      description: Internal Request results
+  steps:
+    - name: create-internal-request
+      image: 
+        quay.io/hacbs-release/release-base-image@sha256:9e7fd1a3ccf0d2c8077f565c78e50862a7cc4792d548b5c01c8b09077e6d23a7
+      script: |
+        #!/usr/bin/env sh
+        PATH=/bin:/usr/bin:/usr/local/bin
+        export PATH
+
+        resourceRequest="/tmp/$$"
+        fbcComponentJson="/tmp/fbc-component-$$.json"
+
+        cat > ${fbcComponentJson} <<JSON
+        $(params.fbcFragment)
+        JSON
+
+        cat > ${resourceRequest} <<YAML
+        ---
+        apiVersion: appstudio.redhat.com/v1alpha1
+        kind: InternalRequest
+        metadata:
+          name: "ir-$(params.pipelineRunName)"
+        spec:
+          request: "iib"
+          params:
+            binaryImage: `printf "%q" '$(params.binaryImage)'`
+            fbcFragment: `jq .components[0].containerImage < ${fbcComponentJson}`
+            fromIndex: $(params.fromIndex)
+            overwriteFromIndex: "$(params.overwriteFromIndex)"
+            buildTags: "$(params.buildTags)"
+            addArches: "$(params.addArches)"
+            buildTimeoutSeconds: `printf "\"%q\"" '$(params.buildTimeoutSeconds)'`
+        YAML
+        kubectl create -f ${resourceRequest}
+    - name: watch-internal-request-status
+      image:
+        quay.io/hacbs-release/release-base-image@sha256:9e7fd1a3ccf0d2c8077f565c78e50862a7cc4792d548b5c01c8b09077e6d23a7
+      script: |
+        #!/usr/bin/env sh
+        PATH=/bin:/usr/bin:/usr/local/bin
+        TASKRUN="/tmp/$$.sh"
+
+        cat > ${TASKRUN} <<SH
```

Comment:
this is in multiple tasks. I think it should be pulled out into its own script and then just called in each task instead of defined in multiple tasks

Reply:
there is another ticket to make the "create-internal-request" generic. This will require a refactoring in this and other tasks related to the (so far) FBC-Release. Then I can change this.

Reply:
@johnbieren it is ok if I do this?

Reply:
Yes I am fine with the change being done in a separate PR